src/collection.ts

```diff
@@ -146,7 +154,7 @@ export interface CollectionPrivate {
  * });
  * ```
  */
-export class Collection {
+export class Collection<TSchema extends Document = any> {
```

Comment:
Everywhere that we use `Collection` it is actually now `Collection<any>`. If we made the change to Document it would default to `Collection<Document>`. It causes a large number of compiler errors, but one example is:

This helper function, that is used in every crud method on collection can no longer accept `getTopology(this)`
```
export function getTopology(provider: MongoClient | Db | Collection): Topology;
```

That is because it accepts Collection<Document> _only_. But if we have some parametrized collection like `Collection<Pet>` then supposedly `getTopology` cannot operate on `Collection<Pet>`. However all that getTopology needs is access to certain properties to detect if its passed a client/db/collection and use that info to find the topology instance. 

There is more errors than just getTopology but its pretty much the same idea with each of them. The solutions I see are the following:
- Leaving Collection to default to `any`, that allows any collection to be passed to these functions that would fail if there was another default type.
- Change Collection to default to Document and make the error spots accept Collection<any> explicitly
- Remove the explicit dependence on Collection when possible and instead require objects that have the correct shape

The last bullet is something like `getTopology` only really depends upon `{s: {db:{s: {client: {toplogy: Topology}}}}}` for the Collection argument.

Reply:
Would using something akin to ` = Partial<Document>` or creating an interim type like `type CollectionDocument = Pick<Document, neededFields> & Partial<Document>`?

Reply:
This turned out to be a lot easier of a fix
```typescript
getTopology<T>(provider: MongoClient | Db | Collection<T>)
```
If we want to have our Collection default to something other than any we just have to allow any generic type to be filled in to helper functions like this. I went through and dismissed the errors generated by the operations layer using `Collection` by using the `as TODO_NODE_2648` cast.